controllers/vrg_kubeobjects.go

```diff
@@ -106,25 +135,105 @@ func (v *VRGInstance) kubeObjectsProtect(captureNumber int64) []error {
 			objectStore.AddressComponent1(),
 			objectStore.AddressComponent2(),
 			v.s3KeyPrefix(),
-			v.instance.Namespace,
-			VeleroNamespaceNameDefault,
+			sourceNamespacedName,
+			captureInstance,
 			captureNumber,
 		); err != nil {
 			v.log.Error(err, "kube object protect", "profile", s3ProfileName)
-			errors = append(errors, err)
+			// don't return error yet since it could be non-fatal (e.g. slow API server)
+		}
+
+		backupNamespacedName := getBackupNamespacedName(sourceNamespacedName.Name,
+			sourceNamespacedName.Namespace, spec.Name, captureNumber)
+
+		// TODO: remove tight loop here
+		backupComplete := false
+		for !backupComplete {
+			backupComplete, err = backupIsDone(v.ctx, v.reconciler.APIReader,
+				objectWriter{v.reconciler.Client, v.ctx, v.log}, backupNamespacedName)
+
+			if err != nil {
+				backupComplete = true
+			}
 		}
 	}
 
-	return errors
+	return err
 }
 
-func (v *VRGInstance) kubeObjectsRecover(objectStore ObjectStorer) error {
-	if v.instance.Spec.KubeObjectProtection == nil {
-		v.log.Info("Kube object recovery disabled")
+// return values: includedResources, excludedResources
+func getTypeSequenceResourceList(toInclude, toExclude []string) ([]string, []string) {
+	included := []string{"*"} // include everything
+	excluded := []string{}    // exclude nothing
+
+	if toInclude != nil {
+		included = toInclude
+	}
+
+	if toExclude != nil {
+		excluded = toExclude
+	}
+
+	return included, excluded
+}
+
+func (v *VRGInstance) getTypeSequenceSubBackups() []ramen.ResourceCaptureInstance {
+	if v.instance.Spec.KubeObjectProtection != nil &&
+		v.instance.Spec.KubeObjectProtection.ResourceCaptureOrder != nil {
```

Comment:
I since the order also specifies scope, I think this field should now be required and not a pointer, same for capture order.  Also can use API server to require length be greater than zero.

Reply:
I think was wrong about this.  The user should be able to leave the order empty and expect all namespace-scoped Kube objects in the namespace as the vrg to be protected and recovered